doc/user/flycheck-versus-flymake.rst

```diff
@@ -315,33 +213,33 @@ to set this variable interactively.
 Multiple syntax checkers per buffer
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-**Flymake** can only use a single syntax checker per buffer.  Effectively, the
-user can only use a single tool to check a buffer, for instance either PHP Mess
-Detector or PHP CheckStyle.  Third party extensions to Flycheck work around this
-limitation by supplying custom shell scripts to call multiple syntax checking
-tools at once.
+**Flymake** will use all the backends added to the
+`flymake-diagnostic-functions` hook to check a buffer; all backends are started
+at the same time, but errors are reported in the buffer as soon as a backend
+returns them.  Also, backends can be written to report errors for the visible
+region of the buffer first, and collect errors for hidden regions later.
 
-**Flycheck** can easily apply multiple syntax checkers per buffer.  For
-instance, Flycheck will check PHP files with PHP CLI first to find syntax
-errors, then with PHP MessDetector to additionally find idiomatic and semantic
-errors, and eventually with PHP CheckStyle to find stylistic errors.  The user
-will see all errors reported by all of these utilities in the buffer.
+**Flycheck** can also apply multiple syntax checkers per buffer, but in sequence
+rather than concurrently.  For instance, Flycheck will check PHP files with PHP
+CLI first to find syntax errors, then with PHP MessDetector to additionally find
+idiomatic and semantic errors, and eventually with PHP CheckStyle to find
+stylistic errors.  The user will see all errors reported by all of these tools
+in the buffer.  However, if the first checker reported at least one error, then
+the subsequent checkers would not be run.
```

Comment:
That's configurable though

Reply:
Sure it is :)

```
‘:next-checkers NEXT-CHECKERS’
     A list denoting syntax checkers to apply after this syntax
     checker, in what we call "chaining" of syntax checkers.

     Each ITEM is a cons cell ‘(LEVEL . CHECKER)’.  CHECKER is a
     syntax checker to run after this syntax checker.  LEVEL is
     an error level.  CHECKER will only be used if there are no
     current errors of at least LEVEL.  LEVEL may also be t, in
     which case CHECKER is used regardless of the current errors.

     ITEM may also be a syntax checker symbol, which is
     equivalent to ‘(t . ITEM)’.

     Flycheck tries all items in order of declaration, and uses
     the first whose LEVEL matches and whose CHECKER is
     registered and can be used for the current buffer.
```

But OK, it might be simpler to describe it this way.

Reply:
Sorry; I meant not documented *in the manual*.  `flycheck.el` itself is well documented, but I have a feeling that not all users will go source spellhunking to find this bit.  Those that do, on the other hand, already know they can tweak practically any package's behavior with a couple lines of Elisp.